tests/components/evohome/test_init.py

```diff
@@ -30,12 +30,21 @@ async def test_entities(
 ) -> None:
     """Test entities and state after setup of a Honeywell TCC-compatible system."""
 
-    # some extended state attrs are relative the current time
+    # Cannot use the evohome fixture, as need to set dtm first
+    #  - some extended state attrs are relative the current time
     freezer.move_to("2024-07-10T12:00:00Z")
 
+    # includes await hass.async_block_till_done(), but is not enough!
     async for _ in setup_evohome(hass, config, install=install):
         pass
 
+        # in async_setup():
+        # -> hass.async_create_task(async_load_platform() x 1-2
+
+        # in async_load_platform(), which calls async_add_entities():
+        # -> ep._async_schedule_add_entities(entities, update_before_add=True)
+        # -> hass.async_create_task_internal(async_add_entities(
+
     assert hass.states.async_all() == snapshot
```

Comment:
Rarely, this test fails because all the entities are not being instantiated before the `hass.states.async_all()`...

I believe this is because the `hass.async_block_till_done()` is returning:
 - _after_ the tasks created by `async_setup()` are finished, but 
 - _before_ the tasks created by `async_load_platform()` calling `sync_add_entities()` with `update_before_add=True`

Adding another `async_block_till_done()` between `setup_evohome()` doesn't solve the issue.

Reply:
```text
------------------------------------------------------------------------------------------- snapshot report summary --------------------------------------------------------------------------------------------
1 snapshot failed. 5 snapshots passed.
=========================================================================================== short test summary info ============================================================================================
FAILED tests/components/evohome/test_init.py::test_entities[botched] - AssertionError: assert [+ received] == [- snapshot]
    list([
  +   StateSnapshot({
  +     'attributes': ReadOnlyDict({
  +       'away_mode': 'on',
  +       'current_temperature': 23,
  +       'friendly_name': 'Domestic Hot Water',
  +       'icon': 'mdi:thermometer-lines',
  +       'max_temp': 60,
  +       'min_temp': 43,
  +       'operation_list': list([
  +         'auto',
  +         'on',
  +         'off',
  +       ]),
  +       'operation_mode': 'off',
  +       'status': dict({
  +         'active_faults': list([
  +         ]),
  +         'dhw_id': '3933910',
  +         'setpoints': dict({
  +           'next_sp_from': '2024-07-10T13:00:00+01:00',
  +           'next_sp_state': 'Off',
  +           'this_sp_from': '2024-07-10T12:00:00+01:00',
  +           'this_sp_state': 'On',
  +         }),
  +         'state_status': dict({
  +           'mode': 'PermanentOverride',
  +           'state': 'Off',
  +         }),
  +         'temperature_status': dict({
  +           'is_available': True,
  +           'temperature': 23.0,
  +         }),
  +       }),
  +       'supported_features': <WaterHeaterEntityFeature: 6>,
  +       'target_temp_high': None,
  +       'target_temp_low': None,
  +       'temperature': None,
  +     }),
  +     'context': <ANY>,
  +     'entity_id': 'water_heater.domestic_hot_water',
  +     'last_changed': <ANY>,
  +     'last_reported': <ANY>,
  +     'last_updated': <ANY>,
  +     'state': 'off',
  +   }),
      StateSnapshot({
         ...
            'setpoint_status': dict({
  -           'setpoint_mode': 'FollowSchedule',
  -           'target_heat_temperature': 17.0,
  -         }),
  -         'setpoints': dict({
  -           'next_sp_from': '2024-07-10T22:10:00+01:00',
  -           'next_sp_temp': 18.6,
  -           'this_sp_from': '2024-07-10T08:00:00+01:00',
  -           'this_sp_temp': 16.0,
  -         }),
  -         'temperature_status': dict({
  -           'is_available': False,
  -         }),
  -         'zone_id': '3432521',
  -       }),
  -       'supported_features': <ClimateEntityFeature: 401>,
  -       'temperature': 17.0,
  -     }),
  -     'context': <ANY>,
  -     'entity_id': 'climate.dead_zone',
  -     'last_changed': <ANY>,
  -     'last_reported': <ANY>,
  -     'last_updated': <ANY>,
  -     'state': 'heat',
  -   }),
  -   StateSnapshot({
  -     'attributes': ReadOnlyDict({
  -       'current_temperature': 19.0,
  -       'friendly_name': 'Main Room',
  -       'hvac_modes': list([
  -         <HVACMode.OFF: 'off'>,
  -         <HVACMode.HEAT: 'heat'>,
  -       ]),
  -       'max_temp': 35.0,
  -       'min_temp': 5.0,
  -       'preset_mode': 'permanent',
  -       'preset_modes': list([
  -         'none',
  -         'temporary',
  -         'permanent',
  -       ]),
  -       'status': dict({
  -         'active_faults': list([
  -           dict({
  -             'faultType': 'TempZoneActuatorCommunicationLost',
  -             'since': '2022-03-02T15:56:01',
    ......

Results (0.78s):
       5 passed
       1 failed
         - tests/components/evohome/test_init.py:23 test_entities[botched]
```